app/models/manageiq/providers/amazon/agent_coordinator_worker.rb

```diff
@@ -0,0 +1,8 @@
+class ManageIQ::Providers::Amazon::AgentCoordinatorWorker < MiqWorker
+  require_nested :Runner
+
+  self.required_roles = ['smartproxy']
```

Comment:
Is this going to run for all smartproxy roles?  That is, if we have a VMware provider will it start one of these?  I think there needs to be more checks here.  cc @agrare

Reply:
Also, isn't this supposed to be one per ems type?  I was expecting the per_ems_type_mixin.

Reply:
If this was per ems type it wouldn't start I think

Reply:
@hsong-rh Can you demo this to me?  I can't see how any of this can work from a full ManageIQ installation.

Reply:
Will change this in new PR, so the coordinator worker only starts when: 
1. smart proxy role enabled;
2. Amazon provider exists in appliance;